builtin/credential/aws/path_config_client.go

```diff
@@ -162,6 +180,41 @@ func (b *backend) pathConfigClientCreateUpdate(
 		configEntry.Endpoint = data.Get("endpoint").(string)
 	}
 
+	iamEndpointStr, ok := data.GetOk("iam_endpoint")
+	if ok {
+		if configEntry.IAMEndpoint != iamEndpointStr.(string) {
+			changedCreds = true
+			configEntry.IAMEndpoint = iamEndpointStr.(string)
+		}
+	} else if req.Operation == logical.CreateOperation {
+		configEntry.IAMEndpoint = data.Get("iam_endpoint").(string)
+	}
+
+	stsEndpointStr, ok := data.GetOk("sts_endpoint")
+	if ok {
+		if configEntry.STSEndpoint != stsEndpointStr.(string) {
+			// We don't directly cache STS clients as they are ever directly used.
+			// However, they are potentially indirectly used as credential providers
+			// for the EC2 and IAM clients, and thus we would be indirectly caching
+			// them there. So, if we change the STS endpoint, we should flush those
+			// cached clients.
+			changedCreds = true
+			configEntry.STSEndpoint = stsEndpointStr.(string)
+		}
+	} else if req.Operation == logical.CreateOperation {
+		configEntry.STSEndpoint = data.Get("sts_endpoint").(string)
+	}
+
+	headerValStr, ok := data.GetOk("iam_server_id_header_value")
+	if ok {
+		if configEntry.IAMServerIdHeaderValue != headerValStr.(string) {
+			// NOT setting changedCreds here, since this isn't really cached
+			configEntry.IAMServerIdHeaderValue = headerValStr.(string)
+		}
+	} else if req.Operation == logical.CreateOperation {
+		configEntry.IAMServerIdHeaderValue = data.Get("iam_server_id_header_value").(string)
+	}
+
 	// Since this endpoint supports both create operation and update operation,
 	// the error checks for access_key and secret_key not being set are not present.
 	// This allows calling this endpoint multiple times to provide the values.
```

Comment:
Below, don't we only need to write the entry `if changedCreds || req.Operation == logical.CreateOperation`?